docs/proposals/conformance-test.md

```diff
@@ -0,0 +1,140 @@
+# Conformance Test for AutoML and Training Working Group
+
+Andrey Velichkevich ([@andreyvelich](https://github.com/andreyvelich))
+Johnu George ([@johnugeorge](https://github.com/johnugeorge))
+2022-11-21
+[Original Google Doc](https://docs.google.com/document/d/1TRUKUY1zCCMdgF-nJ7QtzRwifsoQop0V8UnRo-GWlpI/edit#).
+
+## Motivation
+
+Kubeflow community needs to design conformance program so the distributions can
+become
+[Certified Kubeflow](https://docs.google.com/document/d/1a9ufoe_6DB1eSjpE9eK5nRBoH3ItoSkbPfxRA0AjPIc/edit?resourcekey=0-IRtbQzWfw5L_geRJ7F7GWQ#).
+Recently, Kubeflow Pipelines Working Group (WG) implemented the first version of
+[their conformance tests](https://github.com/kubeflow/kubeflow/issues/6485).
+We should design the same program for AutoML and Training WG.
+
+This document is based on the original proposal for
+[the Kubeflow Pipelines conformance program](https://docs.google.com/document/d/1_til1HkVBFQ1wCgyUpWuMlKRYI4zP1YPmNxr75mzcps/edit#).
+
+## Objective
+
+Conformance program for AutoML and Training WG should follow the same goals as Pipelines program:
+
+- The tests should be fully automated and executable by anyone who has public
+  access to the Kubeflow repository.
+- The test results should be easy to verify by the Kubeflow Conformance Committee.
+- The tests should not depend on cloud provider (e.g. AWS or GCP).
+- The tests should cover basic functionality of Katib and the Training Operator.
+  It will not cover all features.
+- The tests are expected to evolve in the future versions.
+
```

Comment:
- The tests should have a well documented and short list of set-up requirements.
- The tests should install and complete in a relatively short period of time (< 30 minutes) with suggested minimum infrastructure requirements i.e. 3 nodes, 24 vcpu, 64 GB RAM, 500 GB Disk.

Reply:
I am not sure that we can achieve < 30 minutes requirement.
If we are going to run more than 1 Katib Experiment in the future, we might need more time. WDYT @johnugeorge ?
What about Pipelines team @james-jwu ?

Reply:
One more idea - The Katib and Training Operator configuration and tests should make attempts to be integrated with the Pipeline configuration and test configuration.      (My point is that we should try to minimize the conformance testing configuration and resource requirements if/when possible).

Reply:
Pipeline requirement is relatively light. See the below in [setup.yaml](https://github.com/kubeflow/kubeflow/blob/d561429847b6b3f1207261bb169bfef61d7888f1/conformance/1.5/setup.yaml#L26):
     cpu: "2"
     memory: 2Gi
     requests.storage: "5Gi"

It's been a while since I last ran the Pipeline tests, but they are quite fast (<15 min for sure). 

How long does the current Katib and Training tests run?

Reply:
@james-jwu Is resources a mandatory requirement ? We have been running Katib deployment + tests on Github CI which has 2-core CPU and 7G memory.  Since allocated resources are bit tight, we have seen that certain runs have exceeded 30 min limit. However, if we have slightly more CPU resources, we can get it in 30 min easily.

Reply:
Also, the katib's hyperparameter searching doesn't care much how the each training step goes on actually, we could set very-small epochs or very-small nueral network for conformance test's experiments.

Reply:
For the 1st version I think it is okay to require more resources. Jaeyeon's suggestion also sounds great.